### PR TITLE
Disable log message in release build

### PR DESCRIPTION
**Pull request by gaojiaqi7**
Log message is disabled in td-shim release version and reserved in the
td-payload and test-td-payload.

Signed-off-by: Jiaqi Gao <jiaqi.gao@intel.com>

### CODE REVIEW

**Comment by gaojiaqi7** [comment]
https://github.com/confidential-containers/td-shim/issues/348

**Comment by gaojiaqi7** [comment]
hi @jyao1 , current implementation reduces the size of release version image from `143 KB` to `81 KB`. After I deleted all calls to the `log`s, the size of the binary is reduced to `75K`, but this reduction should be due to the string literal, do you think it is neccessary for us to do more optimizations? I am not sure if the string literals can be optimized if we use another macro to replace the `log` with an empty implementation.

**Comment by jodh-intel** [comment]
@gaojiaqi7, @jyao1 - you might want to consider tweaking the features in the manifest file. If you set [`release_max_level_off`](https://docs.rs/log/0.4.17/log/#compile-time-filters), all the logging code would be optimised out at compilation time for release builds, or if that's too brutal, `release_max_level_warn` might work?

**Comment by gaojiaqi7** [comment]
thanks @jodh-intel , [release_max_level_off](https://docs.rs/log/0.4.17/log/#compile-time-filters) is good for us, we need to disable all the log message in release build and optimize the code size as much as possible

**Comment by jyao1** [comment]
I think we need use `release_max_level_off`. Dont use runtime check.

File: td-shim/src/bin/td-shim/main.rs [hunk 1]
```diff
@@ -95,7 +95,13 @@ pub extern "win64" fn _start(
     info: usize,
 ) -> ! {
     // The bootstrap code has setup the stack, but only the stack is available now...
-    let _ = td_logger::init();
+    let log_level = if cfg!(debug_assertions) {
```
Comment (jyao1) [inline td-shim/src/bin/td-shim/main.rs:98]: I think we need use `release_max_level_off`. Dont use runtime check.

Comment (gaojiaqi7) [reply td-shim/src/bin/td-shim/main.rs]: Thanks, I have updated the PR

**Review by jyao1 — APPROVED** [review]

**Pull request merged**